bf2fc6cc711aee1a0c2a/kas-fleet-manager
# feat: add cluster_provider_resource_quota_max_allowed metric

## Description
Adds and exposes `kas_fleet_manager_cluster_provider_resource_quota_max_allowed` metric in the KAS Fleet Manager /metric endpoint. This provides information on the maximum allowed number of resources that can be consumed by the provided cluster provider user (i.e. ocm user). 

We only need to expose metrics for `ocm` cluster provider. OCM quota metrics will be set when the user provides ocm client credentials to the kfm service, either ocm clientid+secret or the token.

Standalone providers doesn't have any resource quota limits so no quota metrics will be exposed. 

Example metric: 
```
kas_fleet_manager_cluster_provider_resource_quota_max_allowed{quota_id="<quota-id>", provider="ocm"} 100
```

## Verification Steps
- [ ] Ensure metric is exposed correctly
    - Run kas fleet manager with ocm credentials specified
    - Curl http://localhost:8080/metrics
    - Ensure the quota metric as defined above are included in the response with correct values

## Checklist (Definition of Done)
